cmd/common/registry.go

```diff
@@ -0,0 +1,377 @@
+			fmt.Fprintf(&out, "\t%s\n", columnName)
+			if description != "" {
+				fmt.Fprintf(&out, "\t  %s\n", description)
+			}
+		}
+	}
+	return out.String()
+}
+
+func buildOutputFormatsHelp(outputFormats gadgets.OutputFormats) []string {
+	var outputFormatsHelp []string
+	var supportedOutputFormats []string
+
+	for ofKey, of := range outputFormats {
+		supportedOutputFormats = append(supportedOutputFormats, ofKey)
+		desc := fmt.Sprintf("%s (%s)", of.Name, ofKey)
+		if of.Description != "" {
+			desc += fmt.Sprintf("\n  %s", of.Description)
+		}
+		outputFormatsHelp = append(outputFormatsHelp, desc)
+	}
+	sort.Strings(outputFormatsHelp)
+	outputFormatsHelp = append([]string{fmt.Sprintf("Output format (%s).", strings.Join(supportedOutputFormats, ", ")), ""}, outputFormatsHelp...)
+	return outputFormatsHelp
+}
+
+func buildCommandFromGadget(
+	gadgetDesc gadgets.GadgetDesc,
+	columnFilters []cols.ColumnFilter,
+	runtime runtime.Runtime,
+	runtimeParams *params.Params,
+	operatorsParamsCollection params.Collection,
+) *cobra.Command {
+	var outputMode string
+	var verbose bool
+	var showColumns []string
+	var filters []string
+	var timeout int
+
+	outputFormats := gadgets.OutputFormats{}
+	defaultOutputFormat := ""
+
+	// Instantiate parser - this is important to do, because we might apply filters and such to this instance
+	parser := gadgetDesc.Parser()
+	if parser != nil && columnFilters != nil {
+		parser.SetColumnFilters(columnFilters...)
+	}
+
+	// Instantiate gadget params - this is important, because the params get filled out by cobra
+	gadgetParams := gadgetDesc.ParamDescs().ToParams()
+
+	// Get per gadget operator params
+	validOperators := operators.GetOperatorsForGadget(gadgetDesc)
+	operatorsParamCollection := validOperators.ParamCollection()
+
+	cmd := &cobra.Command{
+		Use:          gadgetDesc.Name(),
+		Short:        gadgetDesc.Description(),
+		Long:         buildGadgetDoc(gadgetDesc, parser),
+		SilenceUsage: true, // do not print usage when there is an error
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if verbose {
+				log.SetLevel(log.DebugLevel)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := runtime.Init(runtimeParams)
+			if err != nil {
+				return fmt.Errorf("initializing runtime: %w", err)
+			}
+			defer runtime.Close()
+
+			err = validOperators.Init(operatorsParamsCollection)
+			if err != nil {
+				return fmt.Errorf("initializing operators: %w", err)
+			}
+
+			fe := console.NewFrontend()
+			defer fe.Close()
+
+			ctx := fe.GetContext()
+
+			// Handle timeout parameter by adding a timeout to the context
+			if timeout != 0 {
+				tmpCtx, cancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
+				defer cancel()
+				ctx = tmpCtx
+			}
+
+			log := logger.DefaultLogger()
+
+			gadgetCtx := gadgetcontext.New(
+				ctx,
+				"",
+				runtime,
+				gadgetDesc,
+				gadgetParams,
+				operatorsParamCollection,
+				parser,
+				log,
+			)
```

Comment:
The CLI code doesn't need to log anything. Any error here should make fail the command, no?

```suggestion
			gadgetCtx := gadgetcontext.New(
				ctx,
				"",
				runtime,
				gadgetDesc,
				gadgetParams,
				operatorsParamCollection,
				parser,
				logger.DefaultLogger(),
			)
```

Reply:
Yeah, I think it was used to output debug information earlier, but we don't need it anymore.